File: 06/nim/part01.nim
# Day 6: Universal Orbit Map
# https://adventofcode.com/2019/day/6

import tables, strutils, sequtils

let input = open("./inputs/06.txt")
var orbits = initTable[string, string]()

for line in input.lines:
  let planets = line.split({ ')' }).mapIt(strip(it))
  orbits[planets[1]] = planets[0]

var total = 0
for planet in orbits.keys:
  var current = planet
  while orbits.contains(current):
    total += 1
    current = orbits[current]

input.close()

echo total

File: 06/ruby/part01.rb
# Day 6: Universal Orbit Map
# https://adventofcode.com/2019/day/6

input = File.read('./inputs/06.txt')
orbits = {}

input.lines.each do |line|
  a, b = line.split(')').map(&:strip)
  orbits[b] = a
end

count = 0
orbits.each_key do |planet|
  while planet = orbits[planet]
    count += 1
  end
end

puts count
